frontend/ui/widget/imageviewer.lua

```diff
@@ -410,6 +417,16 @@ function ImageViewer:_new_image_wg()
         rotation_angle = rotate_clockwise and 90 or 270
     end
 
+    if self._scaled_image_func then
+        local scale_factor_used
+        self.image, scale_factor_used = self._scaled_image_func(self.scale_factor, max_image_w, max_image_h)
+        if self.scale_factor == 0 then
+            -- onZoomIn/Out need to know the current scale factor, that they won't be
+            -- able to fetch from _image_wg as we force it to be 1. So, remember it.
+            self._scale_factor_0 = scale_factor_used
+        end
+    end
+
```

Comment:
Possible easier to rearrange
```
if self.scale_factor == 0 then
    self.image, self._scale_factor_0 = self._scaled_image_func(self.scale_factor, max_image_w, max_image_h)
end
```

Reply:
I need self.image also when scale_factor is not 0 :)
You can ignore my _scale_factor_0 when thinking about zoom - it is just a needed trick the way we handle SVG.